x/crosschain/keeper/abci.go

```diff
@@ -175,3 +185,55 @@ func CheckAndUpdateCctxGasPrice(
 
 	return gasPriceIncrease, additionalFees, nil
 }
+
+// CheckAndUpdateCctxGasRateBTC checks if the retry interval is reached and updates the gas rate if so
+// Zetacore only needs to update the gas rate in CCTX and fee bumping will be handled by zetaclient
+func CheckAndUpdateCctxGasRateBTC(
+	ctx sdk.Context,
+	k Keeper,
+	cctx types.CrossChainTx,
+	flags observertypes.GasPriceIncreaseFlags,
+) (math.Uint, math.Uint, error) {
+	// skip if gas price or gas limit is not set
+	if cctx.GetCurrentOutboundParam().GasPrice == "" || cctx.GetCurrentOutboundParam().CallOptions.GasLimit == 0 {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// skip if retry interval is not reached
+	lastUpdated := time.Unix(cctx.CctxStatus.LastUpdateTimestamp, 0)
+	if ctx.BlockTime().Before(lastUpdated.Add(flags.RetryInterval)) {
+		return math.ZeroUint(), math.ZeroUint(), nil
+	}
+
+	// compute gas price increase
+	chainID := cctx.GetCurrentOutboundParam().ReceiverChainId
+	medianGasPrice, _, isFound := k.GetMedianGasValues(ctx, chainID)
+	if !isFound {
+		return math.ZeroUint(), math.ZeroUint(), cosmoserrors.Wrap(
+			types.ErrUnableToGetGasPrice,
+			fmt.Sprintf("cannot get gas price for chain %d", chainID),
+		)
+	}
+
+	// set new gas rate and last update timestamp
+	// there is no priority fee in Bitcoin, we reuse 'GasPriorityFee' to store latest gas rate in satoshi/vByte
+	cctx.GetCurrentOutboundParam().GasPriorityFee = medianGasPrice.String()
+	k.SetCrossChainTx(ctx, cctx)
+
+	return math.ZeroUint(), math.ZeroUint(), nil
+}
+
+// GetCctxGasPriceUpdater returns the function to update gas price according to chain type
+func GetCctxGasPriceUpdater(chainID int64, additionalChains []zetachains.Chain) (CheckAndUpdateCctxGasPriceFunc, bool) {
```

Comment:
Why creating this getter function instead of a `CheckAndUpdateCctxGasPrice` that dispatch to the correct function?

Reply:
Created `CheckAndUpdateCCTXGasPrice` to dispatch EVM/BTC gas price updating logic to `CheckAndUpdateCCTXGasPriceEVM` and `CheckAndUpdateCCTXGasPriceBTC`: https://github.com/zeta-chain/node/pull/3306/commits/24d12ae23bb5cd37092394a76b2ef55a8aadce94